### PR TITLE
[DOC] Tweaks for Array#shift

### PR DESCRIPTION
**Pull request by BurdetteLamar**

### CODE REVIEW

File: array.c [hunk 1]
```diff
@@ -1512,35 +1512,38 @@ rb_ary_shift(VALUE ary)
 
 /*
  *  call-seq:
- *     array.shift -> object or nil
- *     array.shift(n) -> new_array
+ *    shift -> new_array or nil
```
Comment (peterzhu2118) [inline array.c:1515]:
```suggestion
 *    shift -> object or nil
```

File: array.c [hunk 2]
```diff
@@ -1512,35 +1512,38 @@ rb_ary_shift(VALUE ary)
...
+ *    shift(count) -> new_array or nil
  *
- *  Removes and returns leading elements.
+ *  Removes and returns leading elements from +self+ in a new array.
```
Comment (peterzhu2118) [inline array.c:1518]: It only returns a new array when `count` is given.

File: array.c [hunk 3]
```diff
@@ -1512,35 +1512,38 @@ rb_ary_shift(VALUE ary)
...
+ *  If +count+ is large,
+ *  removes and returns all elements:
  *
- *    a = [:foo, 'bar', 2]
- *    a.shift(3) # => [:foo, 'bar', 2]
+ *    a = [0, 1, 2, 3]
+ *    a.shift(50) # => [0, 1, 2, 3]
+ *    a           # => []
  *
- *  If +n+ is zero, returns a new empty +Array+; +self+ is unmodified.
```
Comment (peterzhu2118) [inline array.c:1541]: The new documentation has lost the case where `count` is zero.

Comment (BurdetteLamar) [reply array.c:1541]: Example added.

File: array.c (hunk 2 above)
Comment (BurdetteLamar) [reply array.c]: Fixed.

**Review by peterzhu2118 — APPROVED** [review]

**Pull request merged**